src/interpreter/terminator/mod.rs

```diff
@@ -472,23 +475,36 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
                         // Interpreter magic: insert an intermediate pointer, so we can skip the
                         // intermediate function call.
-                        // FIXME: this is a memory leak, should probably add the pointer to the
-                        // current stack.
-                        let first = self.value_to_ptr_dont_use(args[0].0, args[0].1)?;
-                        args[0].0 = Value::ByVal(PrimVal::from_ptr(first));
+                        let ptr = match args[0].0 {
```

Comment:
done (stayed with `Pointer`s, since `allocate` returns a `Pointer` and `deallocate` takes one)